src/server/lib/okta/register.ts

```diff
@@ -211,14 +109,63 @@ export const resendRegistrationEmail = async (
       });
       if (!emailIsSent) {
         throw new OktaError({
-          message: `Okta user activation failed: Failed to send email`,
+          message: `Okta user reset password failed: Failed to send email`,
         });
       }
       return user;
     }
     default:
       throw new OktaError({
-        message: `Okta email resend failed with unaccepted Okta user status: ${user.status}`,
+        message: `Okta registration/resend failed with unaccepted Okta user status: ${user.status}`,
       });
   }
 };
+
+/**
+ * @method register
+ *
+ * Attempts to create a new user in Okta without credentials and send the activation/welcome email.
+ *
+ * If a user already exists, we read the state of the user and take a recovery flow action depending on that.
+ *
+ * @param {string} email
+ * @returns {Promise<UserResponse>} Promise that resolves to the user object
+ */
+export const register = async (email: string): Promise<UserResponse> => {
+  try {
+    return await createUser({
+      profile: {
+        email,
+        login: email,
+        isGuardianUser: true,
+        registrationPlatform: 'identity-gateway',
+      },
+    });
+  } catch (error) {
+    if (
+      error instanceof OktaError &&
+      error.name === 'ApiValidationError' &&
+      causesInclude(error.causes, 'already exists')
+    ) {
+      return handleUserAlreadyExists(email);
+    } else {
+      throw error;
+    }
+  }
+};
+
+/**
+ * @method resendRegistrationEmail
+ *
+ * Used by the email sent page if we need to resend the registration email.
+ *
+ * We read the state of the user and send the appropriate email.
+ *
+ * @param {string} email
+ * @returns {Promise<UserResponse>} Promise that resolves to the user object
+ */
+export const resendRegistrationEmail = async (
```

Comment:
not sure i like the layer of indirection here with the wrapping method.. if both methods do the same job, is there a name that would be able to cover the intent for all use cases instead of having to wrap it like this?

Reply:
Yeah good point. Naming things is hard. 🤔

Reply:
Changed it to `sendRegistrationEmailByUserState`

Reply:
nice 👍